cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java

```diff
@@ -277,53 +276,81 @@ public static String clientsCaCertSecretName(String cluster) {
     }
 
     public static KafkaCluster fromCrd(Kafka kafkaAssembly, KafkaVersion.Lookup versions) {
+        return fromCrd(kafkaAssembly, versions, null);
+    }
+
+    public static KafkaCluster fromCrd(Kafka kafkaAssembly, KafkaVersion.Lookup versions, Storage oldStorage) {
         KafkaCluster result = new KafkaCluster(kafkaAssembly.getMetadata().getNamespace(),
                 kafkaAssembly.getMetadata().getName(),
                 Labels.fromResource(kafkaAssembly).withKind(kafkaAssembly.getKind()));
+
         result.setOwnerReference(kafkaAssembly);
+
         KafkaClusterSpec kafkaClusterSpec = kafkaAssembly.getSpec().getKafka();
+
         result.setReplicas(kafkaClusterSpec.getReplicas());
+
         String image = versions.kafkaImage(kafkaClusterSpec.getImage(), kafkaClusterSpec.getVersion());
         if (image == null) {
             throw new InvalidResourceException("Version " + kafkaClusterSpec.getVersion() + " is not supported. Supported versions are: " + String.join(", ", versions.supportedVersions()) + ".");
         }
         result.setImage(image);
+
         if (kafkaClusterSpec.getReadinessProbe() != null) {
             result.setReadinessInitialDelay(kafkaClusterSpec.getReadinessProbe().getInitialDelaySeconds());
             result.setReadinessTimeout(kafkaClusterSpec.getReadinessProbe().getTimeoutSeconds());
         }
+
         if (kafkaClusterSpec.getLivenessProbe() != null) {
             result.setLivenessInitialDelay(kafkaClusterSpec.getLivenessProbe().getInitialDelaySeconds());
             result.setLivenessTimeout(kafkaClusterSpec.getLivenessProbe().getTimeoutSeconds());
         }
+
         result.setRack(kafkaClusterSpec.getRack());
 
         String initImage = kafkaClusterSpec.getBrokerRackInitImage();
         if (initImage == null) {
             initImage = KafkaClusterSpec.DEFAULT_INIT_IMAGE;
         }
         result.setInitImage(initImage);
+
         Logging logging = kafkaClusterSpec.getLogging();
         result.setLogging(logging == null ? new InlineLogging() : logging);
+
         result.setGcLoggingEnabled(kafkaClusterSpec.getJvmOptions() == null ? true : kafkaClusterSpec.getJvmOptions().isGcLoggingEnabled());
+
         result.setJvmOptions(kafkaClusterSpec.getJvmOptions());
+
         result.setConfiguration(new KafkaConfiguration(kafkaClusterSpec.getConfig().entrySet()));
+
```

Comment:
I never knew you had such strong feelings about blank lines :open_mouth:

Reply:
It makes it a bit more readable for me. I can revert it if you want ;-).

Reply:
FTR MHO is blank lines to separate/group/delineate important bits of code are useful. Blank lines between every statement is harmful because it means I can literally only get half as much code on my screen. Sometimes it is _really helpful_ to have as much code on screen as possible. Also if every statement is separated by a blank line then I suddenly need _two_ blank lines to serve the legitimate separate/group/delineate need, and it just starts getting out of hand.